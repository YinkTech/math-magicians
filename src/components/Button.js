import React from 'react';
import { PropTypes } from 'prop-types';
class Button extends React.Component {
    render() {
        return (
            <button type='button' name={this.props.name} className={this.props.className} onClick={e => this.props.handleClick(e.target.name)} > {this.props.name}</button>
        );
    };
};

Button.propTypes = {
    name: PropTypes.string.isRequired,
    className: PropTypes.string.isRequired,
    handleClick: PropTypes.func.isRequired
};

export default Button;