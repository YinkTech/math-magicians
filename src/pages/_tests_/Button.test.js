import React from "react";
import Button from "./../../components/Button";
import renderer from "react-test-renderer";

it("If it renders correctly", () => {
    const button = renderer
        .create(<Button />)
        .toJSON();
        expect(button).toMatchSnapshot();
});
