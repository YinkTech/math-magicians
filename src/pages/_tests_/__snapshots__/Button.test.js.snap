// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`If it renders correctly 1`] = `
<button
  type="button"
>
   
</button>
`;
